### PR TITLE
Remove changelog and document automated creation

### LINKED ISSUE #3077: Selecting tags for changelog automation
### What is the expected behavior?

As a part of automating more of our daily workflow with https://github.com/mtreinish/qiskit-bot
one of the things that we'll no longer have to do manually is create a changelog file. We write real documentation for releases using reno now and the changelog is mostly overlapping with data in the git log and github PRs so there's no need to manually curate that anymore. qiskit-bot will generate a changelog for us by using the commit msg summary from the squashed merge. It looks for tags in the git history and uses that as categories for the changelog. For example see: https://github.com/mtreinish/qiskit-terra-sandbox/releases/tag/0.16.2 which was generated by the bot based on tagged PRs.

For this to work we need to assign tags to each category heading in the changelog, right now there are defaults set in the bot: https://github.com/mtreinish/qiskit-bot/blob/master/qiskit_bot/config.py#L25-L31 but before we deploy the bot into production we need to pick tags and categories that map with them. We can add new tags or reuse existing ones. It doesn't matter as long as we decide.

One idea I had was add a new class of tags with a `changelog:` prefix to indicate we want this PR included in the changelog. For example adding new tags like: `Changelog: New Feature` and `Changelog: BugFix`.

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

With the creation of the [qiskit-bot](https://github.com/Qiskit/qiskit-bot) release
automation bot, in terra we no longer need to manually maintain or keep 
a changelog in repo. The github release pages (which we were manually copy
and pasting the changelog into already) take the place of this file. For all future
releases the creation of this page is completely automatic and handled
by the bot. This works by using the git log and PR tags to figure out
which commits belong in the changelog summary. The commit msg summary
(the first line) is used for the changelog based on the tags. If a
longer explanation is necessary that should be handled in the reno
release notes.

### Details and comments

Fixes #3077

### CODE REVIEW

**Comment by mtreinish** [comment]
Putting on hold until I configure the production deployment of the bot to report on terra. (right now it's still watching my sandbox repos: https://github.com/mtreinish/qiskit-terra-sandbox/releases and https://github.com/mtreinish/qiskit-sandbox/pulls?utf8=%E2%9C%93&q=is%3Apr+ )

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -100,8 +100,31 @@ please ensure that:
 3. If it makes sense for your change that you have added new tests that
    cover the changes.
 4. Ensure that if your change has an end user facing impact (new feature,
-   deprecation, removal etc) that you have updated the CHANGELOG.md and
-   added a reno release note for that change.
+   deprecation, removal etc) that you have added a reno release note for that
```
Comment (1ucian0) [inline CONTRIBUTING.md]: Maybe not everybody knows what's reno...
```suggestion
   deprecation, removal etc) that you have added a [reno release note](https://docs.openstack.org/reno/latest/) for that
```

**Comment by mtreinish** [comment]
I've configured the bot to watch terra now, so there is no blocker on this anymore

**Comment by mtreinish** [comment]
@kdk sorry for the delay, I've just pushed up the README updates too

**Pull request merged**